ui/pages/Project/components/SavedVariants.jsx

```diff
@@ -142,12 +143,15 @@ class BaseProjectSavedVariants extends React.PureComponent {
       project.variantTagTypes.map(type => type.category).filter(category => category),
     )]
 
-    const isCategory = categoryOptions.includes(newTag)
-    updateTableField('categoryFilter')(isCategory ? newTag : null)
+    const lastNewTag = newTag.length > 0 ? newTag[newTag.length - 1] : null
+    const isCategory = categoryOptions.includes(lastNewTag)
+    const [firstTag, ...otherTag] = newTag
+    const updatedTag = firstTag === ALL_FILTER || categoryOptions.includes(firstTag) ? otherTag : newTag
+    updateTableField('categoryFilter')(isCategory ? lastNewTag : null)
```

Comment:
The behaviors include: if the latest selected option is a category, remove all selected options; otherwise if the previous (or the first) selected option is a category, remove the category option.
The issue is that when a category is selected, the category disappears from the option list.
No category is selected:
<img src=https://github.com/broadinstitute/seqr/assets/33550059/5e1b7312-7f4e-4bbf-8016-3540f5a93943 width=320 height=600 />

After a category (`Collaboration`) is selected:
<img src=https://github.com/broadinstitute/seqr/assets/33550059/ac5096fe-8c12-4814-abdb-0c715c293a85 width=320 height=580 />

Reply:
if a category is selected, all the tags in the category should be shown as selected and should be added to the list, the category itself should remain in the dropdown and clicking it again will have no effect

Reply:
> the category itself should remain in the dropdown and clicking it again will have no effect

How about the category also disappear from the dropdown list? It is much easier to implement and it doesn't make sense to click on a selected category.